src/main/java/com/teradata/tpcds/ScalingInfo.java

```diff
@@ -74,16 +74,24 @@ private long computeCountUsingStaticScale()
         return getRowCountForScale(1);
     }
 
-    private long computeCountUsingLogScale(int scale)
+    private long computeCountUsingLogScale(double scale)
     {
         int scaleSlot = getScaleSlot(scale);
         long delta = getRowCountForScale(DEFINED_SCALES[scaleSlot]) - getRowCountForScale(DEFINED_SCALES[scaleSlot - 1]);
-        float floatOffset = (float) (scale - DEFINED_SCALES[scaleSlot - 1]) / (float) (DEFINED_SCALES[scaleSlot] - DEFINED_SCALES[scaleSlot - 1]);
+        double floatOffset = (scale - DEFINED_SCALES[scaleSlot - 1]) / (DEFINED_SCALES[scaleSlot] - DEFINED_SCALES[scaleSlot - 1]);
```

Comment:
Reverted to float. Although double is a superset of float so it should be fine, but who knows. The C generator is packed with bugs so I agree with you that we should stick to it as close as possible.